src/search/landmarks/landmark_factory_relaxation.h

```diff
@@ -0,0 +1,47 @@
+#ifndef LANDMARKS_LANDMARK_FACTORY_RELAXATION_H
+#define LANDMARKS_LANDMARK_FACTORY_RELAXATION_H
+
+#include "landmark_factory.h"
+
+namespace landmarks {
+class Exploration;
+
+class LandmarkFactoryRelaxation : public LandmarkFactory {
+protected:
+    explicit LandmarkFactoryRelaxation(const options::Options &opts)
+        : LandmarkFactory(opts) {}
+
+    bool relaxed_task_solvable(const TaskProxy &task_proxy, Exploration &exploration,
+                               bool level_out,
+                               const LandmarkNode *exclude,
+                               bool compute_lvl_op = false) const;
+    bool relaxed_task_solvable(const TaskProxy &task_proxy, Exploration &exploration,
+                               std::vector<std::vector<int>> &lvl_var,
+                               std::vector<utils::HashMap<FactPair, int>> &lvl_op,
+                               bool level_out,
+                               const LandmarkNode *exclude,
+                               bool compute_lvl_op = false) const;
+
+private:
+    void generate_landmarks(const std::shared_ptr<AbstractTask> &task) override;
+
+    virtual void generate_landmarks(const std::shared_ptr<AbstractTask> &task,
+                                    Exploration &exploration) = 0;
+    void generate(const TaskProxy &task_proxy, Exploration &exploration);
+
+    void discard_noncausal_landmarks(const TaskProxy &task_proxy,
+                                     Exploration &exploration);
+    bool is_causal_landmark(const TaskProxy &task_proxy,
+                            Exploration &exploration,
+                            const LandmarkNode &landmark) const;
+
+    void calc_achievers(const TaskProxy &task_proxy, Exploration &exploration);
+    bool achieves_non_conditional(const OperatorProxy &o,
```

Comment:
On the original method, there was a comment that this was protected for RPGLandmarks. Does it override the method? If yes, it no longer does. Do your experiments include all landmark graph factories?

Reply:
The comment in the original code was incorrect, neither ```achieves_non_conditional``` nor ```is_landmark_precondition``` is used by any of the 2 landmark factories with "RPG" in their names. The only usages were in ```landmark_factory.h```, and the functions have been moved to ```landmark_factory_relaxation.h``` since that's the only class that uses the functions after the refactoring.